### PR TITLE
create bucket: ignore response body

### PR DESCRIPTION
**Pull request by andreikop**
Without this PRs create_bucket reports failure on some S3 setups even if response is OK.

@bji, you could give me write permissions if you are tired to merge my PRs

### CODE REVIEW

**Comment by andreikop** [comment]
@bji, beep

File: GNUmakefile [hunk 1]
```diff
@@ -136,7 +136,9 @@ ifndef CFLAGS
     endif
 endif
 
-CFLAGS += -Wall -Werror -Wshadow -Wextra -Iinc \
+CFLAGS += -Wall -Werror -Wshadow -Wextra \
+		  -Wno-unused-parameter \
```
Comment (bji) [inline GNUmakefile]: Sorry can you explain (or explain again, if you've already explained it to me and I forgot) why we want -Wno-unused-parameter?  The code was as far as I know clean with respect to this warning, so why do we need to disable the unused-parameter warning now?

**Comment by bji** [comment]
Sorry I'm lazy and only have one eye half-on this project (more like one eye 1/100th on this project).  If you could just address my comment, I'll be happy to merge the pull request.

**Comment by andreikop** [comment]
@bji , check the diff.
It contains function `createBucketFromS3Callback` which doesn't use its parameters by design.
Another solution will be to remove this function and do not generate errors if libs3 got response body but callback for it is not set.
Current solution seems safer for me.

Btw, since this PR was from master but not from a feature-branch, now it contains 1 more commit which I also would like to merge.

**Comment by bji** [comment]
You can spot-fix the one or two places where unused parameters are present like so:

void function(int unused_parameter)
{
    (void) unused_parameter;
}

alternately:

void function (int)
{
}

I would prefer this to disabling that warning altogether.

**Comment by andreikop** [comment]
@bji , done!

**Comment by bji** [comment]
Thank you, very much appreciated.

**Pull request merged**